### PR TITLE
refactor(sidenav): new component

### PR DESCRIPTION
**Pull request by rinaok**
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/nav

### CODE REVIEW

File: libs/components/src/lib/sidenav/README.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
+# sidenav
+
+This element's attributes include the [anchor element](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a) attributes.
+
```
Comment (yinonov) [inline libs/components/src/lib/sidenav/README.md:4]: what?

File: libs/components/src/lib/sidenav/README.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
+# sidenav
```
Comment (yinonov) [inline libs/components/src/lib/sidenav/README.md:1]:
```suggestion
# Side Navigation
```

File: libs/components/src/lib/sidenav/README.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+```html preview
+<vwc-sidenav>
+    <vwc-sidenav-item href="#" text="Profile"></vwc-sidenav-item>
+    <vwc-sidenav-item href="#" text="GitHub"></vwc-sidenav-item>
+    <vwc-sidenav-item href="#" text="lorem ipsum"></vwc-sidenav-item>
+<vwc-sidenav>
+```
+
+### SideNavHeading
```
Comment (yinonov) [inline libs/components/src/lib/sidenav/README.md:25]: maybe this belongs to future side navigation heading page

File: libs/components/src/lib/sidenav/README.md [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+```js
+<script type="module">
+    import '@vonage/vivid/sidenav';
+</script>
+```
+
+## Slots
+
+### SideNavItem
```
Comment (yinonov) [inline libs/components/src/lib/sidenav/README.md:13]: fix all headings to readable titles?
```suggestion
### Side Navigation Item
```

Comment (yinonov) [reply libs/components/src/lib/sidenav/README.md:13]: but this belongs to side navigation page. have a look at calendar and calendar event

File: libs/components/src/lib/sidenav/README.md (hunk 1 above)
Comment (yinonov) [reply libs/components/src/lib/sidenav/README.md:4]: I think this should describe what the custom element is. as in other elements docs. and what it adds semantically

**Review by yinonov — CHANGES_REQUESTED** [review]

**Review by yinonov — COMMENTED** [review]
Maybe break sidenav and sidenav disclosure to 2 different PRs?

**Comment by rinaok** [comment]
> Maybe break sidenav and sidenav disclosure to 2 different PRs?

ok

File: change/@vonage-vivid-eb2e0c3d-1467-441c-a29f-6d551407e326.json [hunk 5]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "none",
+  "comment": "sidenav",
+  "packageName": "@vonage/vivid",
+  "email": "rina.oksman@vonage.com",
+  "dependentChangeType": "none"
```
Comment (yinonov) [inline change/@vonage-vivid-eb2e0c3d-1467-441c-a29f-6d551407e326.json:6]: it's a prerelease. comment can be more explanatory

File: libs/components/src/lib/sidenav-item/sidenav-item.ts [hunk 6]
```diff
@@ -9,7 +9,7 @@ import { TextAnchor } from '../text-anchor/text-anchor';
  *
  * @public
  */
-export class SidenavItem extends TextAnchor {}
+export class SidenavItem extends TextAnchor { }
 
-export interface SidenavItem extends AffixIcon {}
+export interface SidenavItem extends AffixIcon { }
```
Comment (yinonov) [inline libs/components/src/lib/sidenav-item/sidenav-item.ts:14]: lint? 🤔

File: apps/docs/_data/components.json [hunk 7]
```diff
@@ -108,7 +108,8 @@
     "title": "sidenav-item",
     "markdown": "./libs/components/src/lib/sidenav-item/README.md",
     "modules": [
-      "/assets/modules/components/sidenav-item/index.js"
+      "/assets/modules/components/sidenav-item/index.js",
+      "/assets/modules/components/text/index.js"
```
Comment (yinonov) [inline apps/docs/_data/components.json:112]: why the text?

**Review by yinonov — CHANGES_REQUESTED** [review]

File: apps/docs/_data/components.json [hunk 8]
```diff
@@ -204,5 +204,14 @@
...
       "/assets/modules/components/banner/index.js"
     ]
+  },
+  {
+    "title": "sidenav",
+    "markdown": "./libs/components/src/lib/sidenav/README.md",
+    "modules": [
+      "/assets/modules/components/sidenav/index.js",
+      "/assets/modules/components/sidenav-item/index.js",
+      "/assets/modules/components/text/index.js"
```
Comment (yinonov) [inline apps/docs/_data/components.json:214]: why text?

**Review by yinonov — CHANGES_REQUESTED** [review]

File: libs/components/src/lib/sidenav/README.md [hunk 9]
```diff
@@ -0,0 +1,23 @@
+# Side Navigation
+
+Represents a sidenav custom element.
```
Comment (yinonov) [inline libs/components/src/lib/sidenav/README.md:3]:
```suggestion
Represents a side navigation custom element.
```

File: libs/components/src/lib/sidenav/README.md [hunk 10]
```diff
@@ -0,0 +1,23 @@
...
+
+### Side Navigation Item
+
+In order to create a hierarchy of navigation elements, add the `vwc-sidenav-item` elements to the `vwc-sidenav`.
+
+```html preview
+<vwc-sidenav>
+    <vwc-sidenav-item href="#" text="Profile"></vwc-sidenav-item>
+    <vwc-sidenav-item href="#" text="GitHub"></vwc-sidenav-item>
+    <vwc-sidenav-item href="#" text="lorem ipsum"></vwc-sidenav-item>
```
Comment (yinonov) [inline libs/components/src/lib/sidenav/README.md:21]: can u adjust ur workspace to apply indentation tab by 2? not sure why we our lint doesn't operate on that

File: libs/components/src/lib/sidenav/sidenav.spec.ts [hunk 11]
```diff
@@ -0,0 +1,21 @@
...
+
+describe('vwc-sidenav', () => {
+	let element: Sidenav;
+
+	beforeEach(async () => {
+		element = (await fixture(
+			`<${COMPONENT_TAG}></${COMPONENT_TAG}>`
+		)) as Sidenav;
+	});
+
```
Comment (yinonov) [inline libs/components/src/lib/sidenav/sidenav.spec.ts:15]: maybe we should have a11y automated test. include items in sidenav.
see in calendar

**Review by yinonov — CHANGES_REQUESTED** [review]

**Review by yinonov — APPROVED** [review]

**Review by yinonov — DISMISSED** [review]

**Review by yinonov — APPROVED** [review]

**Pull request merged**